### PR TITLE
Introduce LOC dashboard

### PR DESCRIPTION
**Pull request by gdethier**
* LOCs are now displayed as work in progress, waiting or completed for both regular users and LLOs.
* The same component (`LocsDashboard`) is used for both users and LLOs thanks to the new `Locs` data structure.

logion-network/logion-internal#1176

### CODE REVIEW

File: src/loc/dashboard/WorkInProgressLocs.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import LocStatusCell from "src/common/LocStatusCell";
+import Table, { Cell, Column, DateTimeCell, EmptyTableMessage } from "src/common/Table";
+import UserIdentityNameCell from "src/common/UserIdentityNameCell";
+
+export interface Props {
+    locs: LocRequestState[];
+    locDetailsPath: (locId: UUID, locType: LocType) => string;
+}
+
+export default function WorkInProgressLocs(props: Props) {
```
Comment (benoitdevos) [inline src/loc/dashboard/WorkInProgressLocs.tsx:18]: This table replaces, among others, the table `RejectedLocRequests` which displayed the reject reason. Is it still viewable somewhere ?

**Review by benoitdevos — APPROVED** [review]
Great job !!

I have a few minor comments.

Also we have to be sure that all the attributes (reject/void reason and timestamps, etc) that used to be shown in the more detailed tables are now visible somewhere - but I guess it's the tester's job...

File: src/loc/Locs.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        }
+    }
+
+    private isUserWorkInProgress(loc: LocRequestState): boolean {
+        const status = loc.data().status;
+        return status === "DRAFT"
+            || (status === "OPEN" && loc.data().voidInfo === undefined && this.hasUnpublishedItems(loc))
+            || status === "REVIEW_ACCEPTED"
+            || status === "REVIEW_REJECTED";
+    }
```
[the remark this answers is not in the document]
Comment (gdethier) [reply src/loc/Locs.ts:67]: Indeed, I had some issues with that so I ditched the condition but you are right.

File: src/loc/dashboard/WorkInProgressLocs.tsx (hunk 1 above)
Comment (gdethier) [reply src/loc/dashboard/WorkInProgressLocs.tsx:18]: I forgot about that, bringing it back.

**Pull request merged**